server/src/main/java/org/elasticsearch/action/admin/indices/resolve/TransportResolveClusterAction.java

```diff
@@ -0,0 +1,258 @@
+
+    protected void doExecuteForked(Task task, ResolveClusterActionRequest request, ActionListener<ResolveClusterActionResponse> listener) {
+        if (ccsCheckCompatibility) {
+            checkCCSVersionCompatibility(request);
+        }
+        assert task instanceof CancellableTask;
+        final CancellableTask resolveClusterTask = (CancellableTask) task;
+        ClusterState clusterState = clusterService.state();
+        Map<String, OriginalIndices> remoteClusterIndices = remoteClusterService.groupIndices(request.indicesOptions(), request.indices());
+        OriginalIndices localIndices = remoteClusterIndices.remove(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY);
+
+        Map<String, ResolveClusterInfo> clusterInfoMap = new ConcurrentHashMap<>();
+        if (remoteClusterIndices.size() > 0) {
+            final Runnable releaseResourcesOnCancel = () -> {
+                logger.trace("clear index responses on cancellation");
+                clusterInfoMap.clear();
+            };
+
+            // add local cluster info if in scope of the index-expression from user
+            if (localIndices != null) {
+                clusterInfoMap.put(
+                    RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY,
+                    new ResolveClusterInfo(true, false, hasMatchingIndices(localIndices, clusterState), Build.current())
+                );
+            }
+
+            final var finishedOrCancelled = new AtomicBoolean();
+            resolveClusterTask.addListener(() -> {
+                if (finishedOrCancelled.compareAndSet(false, true)) {
+                    releaseResourcesOnCancel.run();
+                }
+            });
+
+            try (RefCountingRunnable refs = new RefCountingRunnable(() -> {
+                finishedOrCancelled.set(true);
+                if (resolveClusterTask.notifyIfCancelled(listener)) {
+                    releaseResourcesOnCancel.run();
+                } else {
+                    listener.onResponse(new ResolveClusterActionResponse(clusterInfoMap));
+                }
+            })) {
+                // make the cross-cluster calls
+                for (Map.Entry<String, OriginalIndices> remoteIndices : remoteClusterIndices.entrySet()) {
+                    resolveClusterTask.ensureNotCancelled();
+                    String clusterAlias = remoteIndices.getKey();
+                    OriginalIndices originalIndices = remoteIndices.getValue();
+                    boolean skipUnavailable = remoteClusterService.isSkipUnavailable(clusterAlias);
+                    Client remoteClusterClient = remoteClusterService.getRemoteClusterClient(
+                        threadPool,
+                        clusterAlias,
+                        searchCoordinationExecutor
+                    );
+                    ResolveClusterActionRequest remoteRequest = new ResolveClusterActionRequest(originalIndices.indices());
+                    remoteRequest.setParentTask(task.getParentTaskId());
```

Comment:
I expect `task` has no parent, so `task.getParentTaskId()` is `TaskId#EMPTY_TASK_ID` which means it won't get cancellations. You need to set the parent task of the remote request to the ID of `task` itself, not its parent.

Reply:
Yes, thanks, that was it. I have it working now with [this commit](https://github.com/elastic/elasticsearch/pull/102726/commits/781a4dd69366b2eb7c4886a9990ed569da71a09c).